### PR TITLE
test: Code Coverage 100% Footer component

### PR DESCRIPTION
**Pull request by CODEX108**
clickUp: https://app.clickup.com/t/85zru69dy

image

### CODE REVIEW

**Review by jayfyle — APPROVED** [review]
looks good to me

File: src/app/shared/components/footer/footer.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,129 @@
...
+    networkServiceSpy.getConnectionStatus.and.returnValue(connectionStatus$);
+    footerComponent.ngOnInit();
+    expect(networkServiceSpy.getConnectionStatus).toHaveBeenCalledTimes(2);
+    expect(footerComponent.connectionState$).toBe(connectionStatus$);
+  });
+
+  describe('goToHome():', () => {
+    it('should emit homeClicked event on clicking home icon', () => {
+      fixture.detectChanges();
+      spyOn(footerComponent.homeClicked, 'emit');
```
Comment (Dimple16) [inline src/app/shared/components/footer/footer.component.spec.ts:67]:
```
const homeClickedSpy = spyOn(footerComponent.homeClicked, 'emit');
expect(homeClickedSpy).toHaveBeenCalledTimes(1);
```

File: src/app/shared/components/footer/footer.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,129 @@
...
+      footerComponent.goToHome();
+      expect(footerComponent.homeClicked.emit).toHaveBeenCalledTimes(1);
+      expect(trackingServiceSpy.footerButtonClicked).toHaveBeenCalledOnceWith({ Action: 'Home', Url: routerSpy.url });
+    });
+
+    it('should navigate to the home page with tasks state if the active state is not HOME', () => {
+      footerComponent.activeState = FooterState.TASKS;
```
Comment (Dimple16) [inline src/app/shared/components/footer/footer.component.spec.ts:74]: shouldn't we add this `fixture.detectChanges();` when assigning `footerComponent.activeState`

File: src/app/shared/components/footer/footer.component.spec.ts [hunk 3]
```diff
@@ -1,26 +1,129 @@
...
+    it('should not navigate if the active state is already HOME', () => {
+      footerComponent.activeState = FooterState.HOME;
+      footerComponent.goToHome();
+      expect(trackingServiceSpy.footerButtonClicked).toHaveBeenCalledOnceWith({ Action: 'Home', Url: routerSpy.url });
+    });
+  });
+
+  it('goToCameraMode(): should emit cameraClicked event on clicking camera icon', () => {
+    fixture.detectChanges();
+    spyOn(footerComponent.cameraClicked, 'emit');
```
Comment (Dimple16) [inline src/app/shared/components/footer/footer.component.spec.ts:88]: same comment, assign it to a variable and reuse

File: src/app/shared/components/footer/footer.component.spec.ts [hunk 4]
```diff
@@ -1,26 +1,129 @@
...
+    it('should disable task click when there are no tasks and the connection status is disconnected', () => {
+      const connectionState = ConnectionMessageStatus.disconnected;
+
+      footerComponent.activeState = FooterState.TASKS;
+      footerComponent.taskCount = 0;
+      footerComponent.goToTasks(connectionState);
+      expect(trackingServiceSpy.footerButtonClicked).not.toHaveBeenCalled();
+    });
+
+    it('should display the task count in the pill when there are tasks', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/shared/components/footer/footer.component.spec.ts:117]: why use `fakeAsync` here?

**Review by Dimple16 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**